src/script/interpreter.cpp

```diff
@@ -1075,10 +1098,16 @@ class CTransactionSignatureSerializer {
             const CTransaction& txTo = dynamic_cast<const CTransaction&>(txBaseTo);
 
             // Serialize vin
-            unsigned int nInputs = fAnyoneCanPay ? 1 : txTo.GetVin().size();
+            // In case of SIGHASH_ANYONECANPAY:
+            // * if Tx has Sc support version and nIn belongs to the CSW inputs - skip inputs serialization
+            // * otherwise only the input being signed is serialized
+            // Otherwise - serialize all inputs
+            unsigned int nInputs = fAnyoneCanPay ?
+                        (txTo.IsScVersion() && nIn >= txTo.GetVin().size() ? 0 : 1) :
+                        txTo.GetVin().size();
```

Comment:
This is a bit hard for me to read. Aren't you say essentially
`if ( ! (txTo.IsScVersion() && nIn >= txTo.GetVin().size()) {/* Do as before*/}
     /*else DO NOTHING*/`?
If so, I would rewrite this as above, for better readability

Reply:
right, i will add `if` check first.

Reply:
not done